spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/ProxyRequestHelper.java

```diff
@@ -129,8 +131,14 @@ public void setResponse(int status, InputStream entity,
 			RequestContext.getCurrentContext().setResponseDataStream(entity);
 		}
 		boolean isOriginResponseGzipped = false;
-		if (headers.containsKey(CONTENT_ENCODING)) {
-			Collection<String> collection = headers.get(CONTENT_ENCODING);
+		if (headers.containsKey(CONTENT_ENCODING) || headers.containsKey(CONTENT_ENCODING_LOWER)) {
```

Comment:
I am looking into this, there is an issue I see. The method you mention receives a HttpServletRequest. The method getHeader of this object is case insensitive. However the setResponse methods receives a MultiValueMap, this does not have the case insensitive get method. Also looking at the HttpHeaders I do not see lowercasing support. Can you me more hints, I like to fix this the right way, but currently I do not see how.